eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -83,25 +79,35 @@ private SubmittedVariantEntity getSubmittedVariantEntity(Document notClusteredSu
 
     @Override
     public void open(ExecutionContext executionContext) throws ItemStreamException {
-        MongoDatabase db = mongoClient.getDatabase(database);
-        MongoCollection<Document> collection = db.getCollection(SUBMITTED_VARIANT_ENTITY);
         Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly),
                 Filters.exists(CLUSTERED_VARIANT_ACCESSION_FIELD, readOnlyClusteredVariants));
         logger.info("Issuing find: {}", query);
-        FindIterable<Document> notClusteredSubmittedVariants = collection.find(query)
-                                                                         .noCursorTimeout(true)
-                                                                         .batchSize(chunkSize);
-        cursor = notClusteredSubmittedVariants.iterator();
+
+        FindIterable<Document> notClusteredSubmittedVariantsDbsnp =
+                getNotClusteredSubmittedVariants(query, DbsnpSubmittedVariantEntity.class);
+        dbsnpCursor = notClusteredSubmittedVariantsDbsnp.iterator();
+        FindIterable<Document> notClusteredSubmittedVariantsEVA =
+                getNotClusteredSubmittedVariants(query, SubmittedVariantEntity.class);
+        evaCursor = notClusteredSubmittedVariantsEVA.iterator();
+
         converter = mongoTemplate.getConverter();
     }
 
+    private FindIterable<Document> getNotClusteredSubmittedVariants(Bson query, Class<?> entityClass) {
```

Comment:
suggest naming this to just **getSubmittedVariants**,  as this will be used to get both clustered as well as non clustered variants based on the query/step